fortran/src/CMakeLists.txt

```diff
@@ -358,7 +359,10 @@ if (BUILD_STATIC_LIBS)
 endif ()
 
 if (BUILD_SHARED_LIBS)
-  add_library (${HDF5_F90_LIBSH_TARGET} SHARED ${f90_F_SOURCES_SHARED})
+  if(MSVC)
+      set(DLLDEF ${HDF5_F90_SRC_BINARY_DIR}/hdf5_fortrandll.def)
+  endif()
+  add_library (${HDF5_F90_LIBSH_TARGET} SHARED ${DLLDEF} ${f90_F_SOURCES_SHARED})
```

Comment:
This relies on "DLLDEF" not being defined for non-msvc. I would prefer that we use an if/else here and in the other file. The other question and maybe why the other form did not work - should we check for it being on windows "if (WIN32) " instead of "if (MSVC)"  - because those are not the same thing. (Another choice is "if (WINDOWS)")